internal/config.go

```diff
@@ -1,19 +1,27 @@
 package internal
 
 import (
-	"gopkg.in/yaml.v2"
+	"errors"
+	"fmt"
 	"os"
+
+	"gopkg.in/yaml.v2"
+)
+
+var (
+	ErrCouldNotReadFile      = errors.New("could not read file")
+	ErrCouldNotUnmarshalYAML = errors.New("could not unmarshal YAML")
 )
 
 func LoadConfig(filename string) (*Config, error) {
 	buf, err := os.ReadFile(filename)
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("%w: %v", ErrCouldNotReadFile, err)
 	}
 
 	var cfg Config
 	if err := yaml.Unmarshal(buf, &cfg); err != nil {
-		return nil, err
+		return nil, fmt.Errorf("%w: %v", ErrCouldNotUnmarshalYAML, err)
```

Comment:
You return two errors here, and you're not wrapping the actual error that caused the problem. 
```suggestion
		return nil, fmt.Errorf("yaml parse failed: %w", err)
```